spec_parser/spec_parser.py

```diff
@@ -254,3 +257,11 @@ def get_text(self, fname: str) -> Optional[str]:
             inp = f.read()
 
         return inp
+
+
+def assign_domain_to_property(specProperty: SpecProperty, classes: List[SpecClass]):
```

Comment:
since this implementation is a loop of all classes and all properties, why not run it once after all properties have been read?

Reply:
I am not sure what you mean, but I admit that I am also a bit confused by the structure of the spec-parser. If we run this after all properties have been read we need to iterate once again through all properties, don't we? 

The `Spec` instance has a namespace containing a list of `SpecClass` instances and `SpecProperty` instances. And each `SpecClass` instance has a dictionary of `properties`.  We want to match each object in the list of `SpecProperty` instances to the keys of the `properties` dict. So if we iterate through all properties of all classes when parsing the properties we save further iterations. Or what would you suggest?

Reply:
After having another look into this I changed the method to directly access the property from the `properties` dict in the `SpecClass` instance which saves one for loop of the properties. I think the implementation makes more sense now.